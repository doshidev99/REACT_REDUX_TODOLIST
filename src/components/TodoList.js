import React from 'react';
import PropTypes from 'prop-types';
import Todo from './Todo';

const TodoList = ({
  todos,
  toggleTodo,
  removeTodo,
  editTodo
}) => (
  <ul>
    {todos.map((todo, index) => (
      <Todo
        key={todo.id}
        {...todo}
        toggleId={() => toggleTodo(todo.id)}
        removeId={() => {
          let array = JSON.parse(
            localStorage.getItem('todos')
          );
          array.splice(index, 1);
          localStorage.setItem(
            'todos',
            JSON.stringify(array)
          );
          removeTodo(todo.id);
        }}
        onSave={text => {
          let array = JSON.parse(
            localStorage.getItem('todos')
          );
          array[index].text = text;
          localStorage.setItem(
            'todos',
            JSON.stringify(array)
          );
          editTodo(todo.id, text);
        }}
      />
    ))}
  </ul>
);

TodoList.propTypes = {
  todos: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      completed: PropTypes.bool.isRequired,
      text: PropTypes.string.isRequired
    }).isRequired
  ).isRequired,
  toggleTodo: PropTypes.func.isRequired
};

export default TodoList;
