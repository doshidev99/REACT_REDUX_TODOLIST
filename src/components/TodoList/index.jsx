import React from 'react';
import PropTypes from 'prop-types';
import Todo from '../Todo';

const TodoList = ({ todos, toggleTodo, removeTodo, editTodo }) => (
  <ul>
    {todos.map((todo, index) => {
      const { text, completed } = todo;
      return (
        <Todo
          key={todo.id}
          text={text}
          completed={completed}
          toggleId={() => toggleTodo(todo.id)}
          removeId={() => {
            const array = JSON.parse(localStorage.getItem('todos'));
            array.splice(index, 1);
            localStorage.setItem('todos', JSON.stringify(array));
            removeTodo(todo.id);
          }}
          onSave={e => {
            const array = JSON.parse(localStorage.getItem('todos'));
            array[index].text = e;
            localStorage.setItem('todos', JSON.stringify(array));
            editTodo(todo.id, e);
          }}
        />
      );
    })}
  </ul>
);

TodoList.propTypes = {
  editTodo: PropTypes.func,
  removeTodo: PropTypes.func,
  todos: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      completed: PropTypes.bool.isRequired,
      text: PropTypes.string.isRequired
    }).isRequired
  ),
  toggleTodo: PropTypes.func
};

TodoList.defaultProps = {
  editTodo: () => {},
  removeTodo: () => {},
  todos: [],
  toggleTodo: () => {}
};

export default TodoList;
