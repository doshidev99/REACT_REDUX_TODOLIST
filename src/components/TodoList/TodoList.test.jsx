/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import { shallow } from 'enzyme';
import TodoList from './index';

describe('TodoList', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<TodoList debug />);

    expect(component).toMatchSnapshot();
  });
});
