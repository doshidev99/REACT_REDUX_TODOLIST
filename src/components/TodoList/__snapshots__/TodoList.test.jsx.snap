// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should render correctly in "debug" mode 1`] = `<ul />`;
